Generate test+train mldata sen1floods11 STAC Catalogs for Raster Vision

CloudNiner opened the pull request:
# Overview

This PR generates a STAC catalog for the s2weak, s1weak, and hand labels sen1floods11 experiments. 

TODO:
- [x] Generate a stac catalog for each of the sen1floods11 experiments: s2weak, s1weak, handlabels, permanentwater
- [x] Allow user to customize test/train split and test/train random seed

# Testing

Follow instructions in Getting Started section of README. A catalog can be created for each experiment described above.

Closes #38

CloudNiner committed:
WIP: First pass sen1floods11 test+train mldata STAC catalog

CloudNiner commented:
@echeipesh @moradology here's a WIP first pass for a quick review to ensure I'm generally on the right track and/or to start building the Raster Vision integration against.

CloudNiner committed:
Parameterize on experiment and test+train split args

moradology commented:
I'm wondering if the top level catalog shouldn't also be a collection: that would allow high level metadata to be easily accessible without risking divergence between `train` and `test`. For example, `channel_order` seems like something that ought to be specified well above the individual items

CloudNiner committed:
Switch to new output schema

Described in:
https://github.com/azavea/noaa-flood-mapping/issues/38#issuecomment-667178703

For this particular catalog, each scene will only
ever have one "labels" link.

CloudNiner commented:
As part of work done for this PR I pushed a RELATIVE_PUBLISHED unzipped stac catalog for sen1floods11 to `s3://sen1floods11-data/catalog/catalog.json`. The original, self-contained catalog is still available at `s3://sen1floods11-data/stac-catalog.zip`.

moradology commented on catalogs/sen1floods11-mldata/main.py:
```diff
@@ -0,0 +1,142 @@
+
+    print("Saving catalog...")
+    mldata_catalog.normalize_hrefs("./data/mldata_{}".format(experiment))
+    import ipdb
+
+    ipdb.set_trace()
```
Looks like some debugging stuck around
